### PR TITLE
Watch sap-btp-operator-controller-manager

### PR DESCRIPTION
**Pull request by MarekMichali**
**Description**

Test scenario 1:

- Patch sap-btp-operator deployment with non existing image
- Delete sap-btp-operator pod
- Wait for sap-btp-operator deployment to be in error
- Wait for sap-btp-operator deployment to be reconciled and ready

Test scenario 2:
- Patch sap-btp-operator-config configmap with random culster_id
- Delete sap-btp-operator pod
- Wait for btpOperator CR to be in error
- Patch back sap-btp-operator-config configmap
- Wait for btpOperator CR to be ready

Changes proposed in this pull request:

- Watch sap-btp-operator-controller-manager state change
- Add 2 E2E tests

**Related issue(s)**

See #913

### CODE REVIEW

File: controllers/btpoperator_controller.go [hunk 1]
```diff
@@ -1343,6 +1348,43 @@ func (r *BtpOperatorReconciler) watchSecretPredicates() predicate.TypedPredicate
...
+			return obj.Name == DeploymentName && obj.Namespace == ChartNamespace
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			obj := e.Object.(*appsv1.Deployment)
+			return obj.Name == DeploymentName && obj.Namespace == ChartNamespace
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			newObj := e.ObjectNew.(*appsv1.Deployment)
+			oldObj := e.ObjectOld.(*appsv1.Deployment)
+			if newObj.Name != DeploymentName || newObj.Namespace != ChartNamespace {
```
Comment (jaroslaw-pieszka) [inline controllers/btpoperator_controller.go:1364]:
```suggestion
			if !(newObj.Name == DeploymentName && newObj.Namespace == ChartNamespace) {
```

**Review by jaroslaw-pieszka — APPROVED** [review]

File: controllers/btpoperator_controller.go (hunk 1 above)
Comment (MarekMichali) [reply controllers/btpoperator_controller.go:1364]: That is how it was originally, but for me the other way round was more readable. We need a 3rd person for this ;)

**Review by szwedm — APPROVED** [review]
/lgtm

**Pull request merged**